packages/iov-keycontrol/src/userprofile.ts

```diff
@@ -184,13 +198,23 @@ export class UserProfile {
     codec: TxCodec,
     nonce: Nonce,
   ): Promise<SignedTransaction> {
+    // identity and the pair (transaction.chainId, transaction.signer) are redundant
+    // but we keep both in the interface to be consistent with appendSignature() where
+    // the original transaction creator is not the signer
+    if (identity.chainId !== transaction.chainId) {
```

Comment:
Good check!

Interesting comment above. In a future refactor, removing WalletId as you mentioned, we could make this `signTransaction(transaction, codec, nonce)`. I like that.

Append signature clearly needs to have a specified identity, but if we can remove it here, I prefer it. (Issue for the future)

Reply:
Jap, I think removing wallet ID and identity are indepeendent and can both be done as separate steps. https://github.com/iov-one/iov-core/issues/641